### PR TITLE
Remove code duplication between returndatacopy and others

### PR DESCRIPTION
**Pull request by xiaodino**
### Description

Remove code duplication between returndatacopy and others as they use similar test codes to construct bytecode for testing.

### Issue Link

https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1324

### Type of change

- [x] Refactor code

### Contents

Remove code duplication in unit tests between returndatacopy, balance, calldataload and others. The unit tests should behave the same.

### Rationale

It's easier to manage the mock bytecode when we move similar code into one place.

### How Has This Been Tested?

Use the command `make test` to run unit tests 

<hr>

### CODE REVIEW

**Review by leolara — APPROVED** [review]
LGTM but it needs a second review :-)

**Review by han0110 — APPROVED** [review]
LGTM! Nice refactor!

**Comment by leolara** [comment]
@xiaodino you can merge when you want

**Pull request merged**
